libkbfs/disk_limiter.go

```diff
@@ -1,37 +1,54 @@
 package libkbfs
 
 import (
+	"fmt"
+
 	"github.com/keybase/client/go/protocol/keybase1"
 	"golang.org/x/net/context"
 )
 
-type diskBlockCacheLimiter interface {
-	// onDiskBlockCacheDelete is called by the disk block cache after deleting
-	// blocks from the cache.
-	onDiskBlockCacheDelete(ctx context.Context, blockBytes int64)
+type diskLimitTrackerType int
 
-	// beforeDiskBlockCachePut is called by the disk block cache before putting
-	// a block into the cache. It returns the total number of available bytes.
-	beforeDiskBlockCachePut(ctx context.Context, blockBytes int64) (
-		availableBytes int64, err error)
+type unknownTrackerTypeError struct {
+	typ diskLimitTrackerType
+}
 
-	// afterDiskBlockCachePut is called by the disk block cache after putting
-	// a block into the cache. It returns how many bytes it acquired.
-	afterDiskBlockCachePut(ctx context.Context, blockBytes int64,
-		putData bool)
+func (e unknownTrackerTypeError) Error() string {
+	return fmt.Sprintf("Unknown tracker type: %d", e.typ)
+}
 
-	// onDiskBlockCacheEnable is called when the disk block cache is enabled to
-	// begin accounting for its blocks.
-	onDiskBlockCacheEnable(ctx context.Context, cacheBytes int64)
+const (
+	unknownLimitTracker diskLimitTrackerType = iota
+	journalLimitTracker
+	diskCacheLimitTracker
+	syncCacheLimitTracker
+)
 
-	// onDiskBlockCacheDisable is called when the disk block cache is disabled to
-	// stop accounting for its blocks.
-	onDiskBlockCacheDisable(ctx context.Context, cacheBytes int64)
+type diskLimitByteTracker interface {
```

Comment:
Yeah I had that kind of refactor halfway done and it only made things worse rather than better, so I rolled back to what we see here. I'll rename to `simpleByteTracker`.